src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+		
+		// Retrieve the SysAdmin Profile Id
+		Id sysAdminId = retrieveSysAdminProfileId();
+
+		// Determine if the User is a SysAdmin
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(sysAdminId == currentUser.ProfileId) {
+					userIdToIsSysAdmin.put(currentUser.Id, true);
+				} else {
+					userIdToIsSysAdmin.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsSysAdmin;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is active based on IsActive field.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsActive status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isActiveUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsActive = new Map<Id, Boolean>();
+		
+		// Determine if the User isActive
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(currentUser.IsActive) {
+					userIdToIsActive.put(currentUser.Id, true);
+				} else {
+					userIdToIsActive.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsActive;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is frozen based on IsFrozen field.
+    * @param updatedUsers The user(s) that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsFrozen status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isFrozenUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsFrozen = new Map<Id, Boolean>();
+
+		// Retrieve the UserLogin records for the updatedUsers
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			Map<Id, Boolean> userLoginUserIdToIsFrozen = retrieveUserLogins(updatedUsers);
+			
+			// Populate the Map with UserId and IsFrozen
+			for(User currentUser : updatedUsers) {
+				userIdToIsFrozen.put(currentUser.Id, userLoginUserIdToIsFrozen.get(currentUser.Id));
+			}
+		}
+
+		return userIdToIsFrozen;
+	}
+	
+
+	/*********************************************************************************************************
+    * @description Handles User changes to IsActive/IsFrozen status by updating the Error Notification Recipients, logging an Error__c, and notifying the org's System Administrators.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	@future
```

Comment:
why the need to use @Future?  we have a naming convention that all of our Future methods should end with the work Future, to make it more obvious to callers.

Reply:
The @future is used to handle the MIXED_DML_OPERATION that occurs when updating the User's isActive field along with other dml operations.

Reply:
could you put that info in the description, as it is very helpful to know why a future needed to be used.  thanks.